pipenv/utils.py

```diff
@@ -1286,7 +1286,9 @@ def venv_resolve_deps(
         os.environ["PIPENV_VERBOSITY"] = str(environments.PIPENV_VERBOSITY)
         os.environ["PIPENV_REQ_DIR"] = fs_str(req_dir)
         os.environ["PIP_NO_INPUT"] = fs_str("1")
-        os.environ["PIPENV_SITE_DIR"] = get_pipenv_sitedir()
+        pipenv_site_dir = get_pipenv_sitedir()
+        if pipenv_site_dir is not None:
+            os.environ["PIPENV_SITE_DIR"] = pipenv_site_dir
```

Comment:
This would fail to unset `PIPENV_SITE_DIR` if the result is None. Probably need an `else` block.

Reply:
Thanks for the review.

Not sure if I follow  - the behaviour is the same as for `keep_outdated` down below this line. Unsetting the environment is done based on the context manager above that keeps a backup. What would be expected in the `else` block?

Reply:
I would expect it to unset the value, so something like `os.environ.pop("PIPENV_SITE_DIR", None)`

Reply:
Done in 5e767e9c. Thanks!